File: TreeNavigator_DataBinding_VB/WF_TreeNavigatorDirectory/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Data
Imports System.Linq
Imports System.Windows.Forms
Imports Syncfusion.Windows.Forms.Tools
Imports System.Data.SqlServerCe

Namespace WF_TreeNavigatorDirectory
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
			dataBaseView = GetOrderDetials()
		End Sub

		Private dataBaseView As DataView

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			Me.treeNavigator1.Visible = True
			FillItems(Me.treeNavigator1)
		End Sub
		Private Sub FillItems(ByVal tree As TreeNavigator)
		  For Each col As DataColumn In dataBaseView.Table.Columns
				Me.treeNavigator1.Items.Add(New TreeMenuItem() With {.Text = col.ColumnName})
		  Next col
			For Each x As DataRow In dataBaseView.Table.Rows
				For Each y As TreeMenuItem In Me.treeNavigator1.Items
					If dataBaseView.Table.Columns.Contains(y.Text) Then
						y.Items.Add(New TreeMenuItem() With {.Text = x(dataBaseView.Table.Columns(y.Text)).ToString()})
					End If

				Next y
			Next x
		End Sub

		Private Shared Function GetOrderDetials() As DataView
			Try
				Dim ds As New DataSet()
				Using con As New SqlCeConnection("DataSource=Data\Northwind.sdf")
					con.Open()
					Dim cmd As New SqlCeCommand("SELECT Top(20) [Order ID], [Product ID], [ColumnName], [Unit Price], Quantity" & ControlChars.CrLf & "                                                          FROM [Order Details] ", con)
					Dim da As New SqlCeDataAdapter(cmd)


					da.Fill(ds)

					'Creation of DataView
					Dim dataView As New DataView(ds.Tables(0), "[Unit Price] >= 20", "[Order ID]", DataViewRowState.CurrentRows)
					Return dataView
				End Using


			Catch e As Exception
				MessageBox.Show("Exception in connection: " & Constants.vbTab + e.Message)
				Return Nothing
			End Try
		End Function
	End Class
End Namespace

File: TreeNavigator_DataBinding_Csharp/TreeNavigator_DataBinding/Form1.cs
using EnvDTE;
using Syncfusion.Windows.Forms.Tools;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Data.SqlServerCe;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace TreeNavigator_DataBinding
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            dataBaseView = GetOrderDetials();
        }

        //Initialize the DataView
        private DataView dataBaseView;

        private void FillItems(TreeNavigator tree)
        {
            foreach (DataColumn col in dataBaseView.Table.Columns)
                this.treeNavigator1.Items.Add(new TreeMenuItem() { Text = col.ColumnName });
            //Add the DataView rows to TreeMenuItem
            foreach (DataRow x in dataBaseView.Table.Rows)
            {
                foreach (TreeMenuItem y in this.treeNavigator1.Items)
                {
                    if (dataBaseView.Table.Columns.Contains(y.Text))
                        y.Items.Add(new TreeMenuItem() { Text = x[dataBaseView.Table.Columns[y.Text]].ToString() });
                }
            }
        }

        private static DataView GetOrderDetials()
        {
            try
            {
                //Initialize the DataSet
                DataSet ds = new DataSet();
                using (SqlCeConnection con = new SqlCeConnection(@"DataSource=Data\Northwind.sdf"))
                {
                    con.Open();
                    SqlCeCommand cmd = new SqlCeCommand("SELECT Top(20) [Order ID], [Product ID], [ColumnName], [Unit Price], Quantity \r\nFROM [Order Details] ", con);
                    SqlCeDataAdapter da = new SqlCeDataAdapter(cmd);


                    da.Fill(ds);

                    // Creation of DataView
                    DataView dataView = new DataView(ds.Tables[0], "[Unit Price] >= 20", "[Order ID]", DataViewRowState.CurrentRows);
                    return dataView;
                }
            }

            catch (Exception e)
            {
                MessageBox.Show("Exception in connection: " +  e.Message);
                return null;
            }
        }


        private void button1_Click(object sender, EventArgs e)
        {
            this.treeNavigator1.Visible = true;
            FillItems(this.treeNavigator1);
        }
    }
}
